### PR TITLE
RavenDB-12110: Failed queries data are now logged in requestLatencyDetails and TrafficWatch.

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

File: src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs [hunk 1]
```diff
@@ -37,36 +49,69 @@ public IndexQueryServerSide(string query, BlittableJsonReaderObject queryParamet
...
             if (result.PageSize == 0 && json.TryGet(nameof(PageSize), out int _) == false)
                 result.PageSize = int.MaxValue;
 
             if (string.IsNullOrWhiteSpace(result.Query))
-                throw new InvalidOperationException($"Index query does not contain '{nameof(Query)}' field.");
+            {
+                var errorMessage = $"Index query does not contain '{nameof(Query)}' field.";
+
+                if (TrafficWatchManager.HasRegisteredClients)
+                    AddStringToHttpContext(httpContext, errorMessage, TrafficWatchChangeType.Queries);
```
Comment (ayende) [inline src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs]: Why not simply wrap the whole Query operation in a `try/catch` ?

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs (hunk 1 above)
Comment (garayx) [reply src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs]: done

Comment (ppekrol) [reply src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs]: Can you push it?

File: src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs [hunk 2]
```diff
@@ -37,85 +49,132 @@ public IndexQueryServerSide(string query, BlittableJsonReaderObject queryParamet
...
+                    tracker.Query = result.Query;
 
-            if (cache.TryGetMetadata(result, out var metadataHash, out var metadata))
-            {
-                result.Metadata = metadata;
                 return result;
             }
+            catch
+            {
+                errorMessage = errorMessage ?? (result == null ? $"Failed to parse index query : {json}" : $"Failed to parse query: {result.Query}");
```
Comment (ppekrol) [inline src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs]: why can't we take the message from the catched exception? We do not need the 'errorMessage' variable here.

Comment (garayx) [reply src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs]: I want to display the Failed query to tracker and trafficwatch, not the exception it throws, also I treat few scenarios here when and log them to trafficwatch and tracker.

Comment (ppekrol) [reply src/Raven.Server/Documents/Queries/IndexQueryServerSide.cs]: Please ping me on Slack

**Review by ppekrol — APPROVED** [review]

**Pull request merged**